### DISCUSSION TITLE
Packages are there own dependents

### DISCUSSION THREAD

**Discussion by dschrempf** [flora-pm/flora-server#207] [Questions] [answered]
Hi!

Thank you for this excellent website! Especially the reverse dependency feature is exciting! (Did you think about package metrics such as download rate? https://crates.io does a good job in that respect, I think.)

I saw that packages show up as their own "Dependents" (should we use "Reverse dependencies"?). For example, the package `circular` shows up in its dependents here: https://flora.pm/packages/@hackage/circular.

I was a bit surprised about this feature! Is this behavior desired?

Thank you!

**Comment by tchoutri** [comment, marked as answer]
Hi @dschrempf!

1. Did you think about package metrics such as download rate?

Yes I have, and it would power the next version of the search engine. See https://github.com/flora-pm/flora-server/issues/4

2. Is this behavior desired?

Looks to me like this should be a custom "ignore" rule, because Flora actually scans all the components of all the releases, and you'll see that your test suite actually depends on `circular`: https://github.com/dschrempf/circular/blob/master/circular.cabal#L51 :) 

So, I don't think we should filter out every `test-suite` component, but we certainly can filter out the package name itself from its own dependents. :)

**Comment by dschrempf** [comment]
It is great that you are working on package metrics!

Yes the test suite dependency is correct: The test suite depends on the library in the package. However, this is most likely the case for all packages.

In conclusion, filtering out the package seems like a good idea.

In the longer run, I think one should distinguish between test or benchmark suite dependencies and dependents and library (or application) dependencies and dependents. For example, take the approach of `crates.io`. E.g., https://crates.io/crates/rand/0.8.5/dependencies. The actual dependencies are separate from the "Dev-Dependencies".

**Reply by tchoutri** [reply]
I've opened https://github.com/flora-pm/flora-server/issues/208 and there is an adjacent ticket here about how to organise them : https://github.com/flora-pm/flora-server/issues/82

**Comment by tchoutri** [comment]
@dschrempf 

> It is great that you are working on package metrics!

Not gonna lie, it's heavy work and I'll need all the help I can get on that. :)

**Comment by tchoutri** [comment]
Update: As of [v1.0.6](https://github.com/flora-pm/flora-server/releases/tag/v1.0.6), packages are no longer shown as their own dependents.
